nodes/src/nodes/context_optimizer/IInstance.py

```
@@ -0,0 +1,111 @@
+
+        # Convert documents to optimizer format
+        docs = []
+        for doc in question.documents or []:
+            doc_dict = doc.model_dump() if hasattr(doc, 'model_dump') else doc.dict()
+            content = doc_dict.get('page_content', '')
+            docs.append({'content': content, '_original': doc})
+
+        # Convert history to optimizer format
+        history = []
+        for h in question.history or []:
+            history.append({'role': h.role, 'content': h.content})
```

Ruff (0.15.7)

[warning] Use a list comprehension to create a transformed list

(PERF401)